util/src/main/java/tc/oc/pgm/util/tablist/TabRender.java

```diff
@@ -146,18 +160,22 @@ public void refreshEntry(TabEntry entry, int index) {
   }
 
   public void updateEntry(TabEntry entry, int index) {
-    this.updatePacket.b.add(
-        NMSHacks.playerListPacketData(
-            this.updatePacket, entry.getId(), this.getContent(entry, index)));
+    getBField(this.updatePacket)
+        .add(
+            NMSHacks.playerListPacketData(
+                this.updatePacket, entry.getId(), this.getContent(entry, index)));
   }
 
   public void updatePing(TabEntry entry, int index) {
-    this.updatePingPacket.b.add(
-        NMSHacks.playerListPacketData(this.updatePingPacket, entry.getId(), entry.getPing()));
+    getBField(this.updatePingPacket)
+        .add(NMSHacks.playerListPacketData(this.updatePingPacket, entry.getId(), entry.getPing()));
   }
 
   public void setHeaderFooter(BaseComponent[] header, BaseComponent[] footer) {
-    view.getViewer().setPlayerListHeaderFooter(header, footer);
+    BungeeComponentSerializer serializer = BungeeComponentSerializer.get();
+    Audience.get(view.getViewer())
+        .sendPlayerListHeaderAndFooter(
+            serializer.deserialize(header), serializer.deserialize(footer));
```

Comment:
This will be very inefficient. We're going `Component` -> `BaseComponent` -> `Component` -> `BaseComponent` -> `String`. Can we just keep the old way of doing it?

Reply:
Well I think the alternative without doing a fair bit of refactoring is 
```
    if (BukkitUtils.isSportPaper()) {
      view.getViewer().setPlayerListHeaderFooter(header, footer);
    } else {
      PacketPlayOutPlayerListHeaderFooter packet = new PacketPlayOutPlayerListHeaderFooter();
      ReflectionUtils.setField(
          packet.getClass(),
          packet,
          IChatBaseComponent.ChatSerializer.a(
              net.md_5.bungee.chat.ComponentSerializer.toString(header)),
          "a");
      ReflectionUtils.setField(
          packet.getClass(),
          packet,
          IChatBaseComponent.ChatSerializer.a(
              net.md_5.bungee.chat.ComponentSerializer.toString(footer)),
          "b");
      send(packet);
    }
```
Not sure which is better

Reply:
Ignore above, refactoring was easier than I though.